tests/Instrumentation/TestActivityListener.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using NUnit.Framework;
+using OpenAI.Chat;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace OpenAI.Tests.Instrumentation;
+
+internal class TestActivityListener : IDisposable
+{
+    private readonly ActivityListener _listener;
+    private readonly ConcurrentQueue<Activity> stoppedActivities = new ConcurrentQueue<Activity>();
+
+    public TestActivityListener(string sourceName)
+    {
+        _listener = new ActivityListener()
```

Comment:
if the tests can take OTel dependency sdk, then consider using InMemoryExporter for these tests.

Reply:
it's possible, are there any specific benefits? I don't mind writing validation against `ActivityListener` since it makes things really easy to debug and understand.